src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,74 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { FC } from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+import { FilterStatus } from '../../types/Filter';
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+type Props = {
+  filterStatus: FilterStatus,
+  setFilterStatus: (status: FilterStatus) => void,
+  filterQuery: string,
+  setFilterQuery: (query: string) => void,
+};
+
+export const TodoFilter: FC<Props> = ({
+  filterStatus,
+  setFilterStatus,
+  filterQuery,
+  setFilterQuery,
+}) => {
+  const filterByOption = Object.entries(FilterStatus);
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={filterStatus}
+            onChange={(event) => setFilterStatus(
+              event.target.value as FilterStatus,
+            )}
+          >
+            {filterByOption.map(([label, value]) => {
+              return (
+                <option
+                  value={value}
+                  key={value}
+                >
+                  {label}
+                </option>
+              );
+            })}
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={filterQuery}
+          onChange={(event) => (
+            setFilterQuery(event.target.value)
+          )}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {filterQuery && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={() => setFilterQuery('')}
```

Comment:
create separate function to clear, maybe in the future you want to clear something else, this will make sense, but it's optional